src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb

```diff
@@ -3551,7 +3550,8 @@ Bailout:
                 ' Now see if any candidates are ambiguous or lose against other candidates in the quickInfo array.
                 ' This loop is destructive to the content of the quickInfo, some applicable candidates could be replaced with
                 ' a "better" candidate, even though that candidate is not applicable, "losers" are deleted, etc.
-                For k As Integer = 0 To If(applicableCount > 0 OrElse Not includeEliminatedCandidates, applicableCount, quickInfo.Count) - 1
+                Dim ke As Integer = If(applicableCount > 0 OrElse Not includeEliminatedCandidates, applicableCount, quickInfo.Count) - 1
+                For k As Integer = 0 To ke
```

Comment:
What is the rational behind this change? Please either revert it or use descriptive name for the local.